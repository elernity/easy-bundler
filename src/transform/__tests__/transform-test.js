'use strict';
require('babel-register')();

jest.mock('../../utils/AssetUtil');

const path = require('path');
const generate = require('@babel/generator').default;
const transform = require('../transform');

const projectRoot = path.join(__dirname, '..');
const context = {
    isImageFile: (filePath) => true,
    addAssetInfo: () => {},
};
const userConfig = {
    projectRoot: projectRoot,
    dev: true,
    platform: 'android',
};

const generateCode = (ast) => {
    return generate(
        ast,
        {
          comments: false,
          compact: false,
          retainLines: false,
        }
    ).code;
}

describe('语法转换测试', () => {

    it('应转换script代码', () => {
        const transformedAst = transform(context, userConfig, path.join(projectRoot, '__files__', 'script.js'), 'script');
        const transformedCode = generateCode(transformedAst);
        expect(transformedCode).toMatchSnapshot();
    });

    it('应转换module代码', () => {
        const transformedAst = transform(context, userConfig, path.join(projectRoot, '__files__', 'module.js'), 'module');
        const transformedCode = generateCode(transformedAst);
        expect(transformedCode).toMatchSnapshot();
    });

    it('应转换asset代码', () => {
        const transformedAst = transform(context, userConfig, path.join(projectRoot, '__files__', 'asset.png'), 'asset');
        const transformedCode = generateCode(transformedAst);
        expect(transformedCode).toMatchSnapshot();
    });

    it('应转换json代码', () => {
        const transformedAst = transform(context, userConfig, path.join(projectRoot, '__files__', 'info.json'), 'json');
        expect(transformedAst).toBeNull();
    });
});
