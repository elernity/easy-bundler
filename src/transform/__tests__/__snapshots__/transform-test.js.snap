// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`语法转换测试 应转换asset代码 1`] = `
"module.exports = require(\\"AssetRegistry.js\\").registerAsset({
  \\"__packager_asset\\": true,
  \\"httpServerLocation\\": \\"\\",
  \\"width\\": 1,
  \\"height\\": 1,
  \\"scales\\": [],
  \\"hash\\": \\"hash\\",
  \\"name\\": \\"name\\",
  \\"type\\": \\"type\\"
});"
`;

exports[`语法转换测试 应转换module代码 1`] = `
"function output() {
  console.log('module');
}

module.exports = output;"
`;

exports[`语法转换测试 应转换script代码 1`] = `"console.log('script');"`;
